interchain/tests/src/interchain_accounts.rs

```diff
@@ -43,61 +42,77 @@ pub fn create_test_remote_account<Chain: IbcQueryHandler, IBC: InterchainEnv<Cha
             namespace: None,
         },
         abstract_core::objects::gov_type::GovernanceDetails::Monarchy {
-            monarch: origin.account.manager.get_chain().sender().to_string(),
+            monarch: abstr_origin
+                .version_control
+                .get_chain()
+                .sender()
+                .to_string(),
         },
-        None,
+        funds.as_deref(),
     )?;
 
-    // We need to register the ibc client as a module of the manager (account specific)
-    origin
-        .account
-        .manager
-        .install_module::<Empty>(IBC_CLIENT, None, None)?;
+    // We need to enable ibc on the account.
+    origin_account.manager.update_settings(Some(true))?;
```

Comment:
To get this to work I had to change 
```
let ibc_client_addr = ibc_client_module.module.reference.unwrap_native()?;
```
to 
```
let ibc_client_addr = ibc_client_module.module.reference.unwrap_adapter()?;
```
Line link: https://github.com/AbstractSDK/abstract/blob/main/framework/contracts/account/manager/src/commands.rs#L818

I figured this out from @Buckram123 's ongoing work: https://github.com/AbstractSDK/abstract/commit/a7c3e7ef3ed6f659b8146484ed5e1be271f1dc0e#diff-ee444462bd0d877187891abac12f00614c31169a132eb8550240a25812ac4287R818

I did not push the change here myself since I am not sure what the desired state should be.

Reply:
It should not be an adapter. It should be native and the registration process needs to be updated to reflect that. I.e. the `get_contracts` function of `Abstract` should include the ibc-client contract. That will ensure it is registered as a native contract.

Reply:
> And that module registration call should be removed

I already applied these changes here: https://github.com/AbstractSDK/abstract/blob/cc2c727bb7fac7accc6f08f7824d6b3723a881c7/framework/packages/abstract-interface/src/deployment.rs#L253-L254

Which is part of why I suggested rebasing to the branch once it compiles.

@Buckram123 Could you revert the unwrap change here: https://github.com/AbstractSDK/abstract/blob/main/framework/packages/abstract-interface/src/deployment.rs#L97